src/collision/collider/mod.rs

```diff
@@ -42,16 +45,167 @@ pub trait IntoCollider<C: AnyCollider> {
     fn collider(&self) -> C;
 }
 
+/// Context necessary to calculate [`ColliderAabb`]s for an [`AnyCollider`]
+#[derive(Deref)]
+pub struct AabbContext<'a, 'w, 's, T: ReadOnlySystemParam> {
+    /// The entity for which the aabb is being calculated
+    pub entity: Entity,
+    #[deref]
+    item: &'a SystemParamItem<'w, 's, T>,
+}
+
+impl<T: ReadOnlySystemParam> Clone for AabbContext<'_, '_, '_, T> {
+    fn clone(&self) -> Self {
+        Self {
+            entity: self.entity,
+            item: self.item,
+        }
+    }
+}
+
+impl<'a, 'w, 's, T: ReadOnlySystemParam> AabbContext<'a, 'w, 's, T> {
+    /// Construct an [`AabbContext`]
+    pub fn new(entity: Entity, item: &'a <T as SystemParam>::Item<'w, 's>) -> Self {
+        Self { entity, item }
+    }
+}
+
+impl AabbContext<'_, '_, '_, ()> {
+    fn fake() -> Self {
```

Comment:
Maybe `empty` would be a better name? Though I guess `Entity::PLACEHOLDER` isn't necessarily "empty"